web/src/components/map/index.js

```diff
@@ -29,11 +30,12 @@ import { Grid } from "../ui";
  */
 export const Map = () => {
   const location = useLocation(); // location for the url
-  const initialDate = moment(location?.state?.date) || moment(); // Date either from upload or current day
+  const initialDate = moment(location?.state?.date); // Date either from upload or current day (moment(undefined) gives now)
   const [mapDate, setMapDate] = useState(initialDate);
   const [formattedDate, setFormattedDate] = useState(
     initialDate.format("YYYY-MM-DD")
   );
+  const [setOfDates, setSetOfDates] = useState(() => new Set()); // Lazy initial render
```

Comment:
TIL about lazy initial states. I have a few situations at work where this will come in handy.